lib/_http_client.js

```diff
@@ -442,16 +442,26 @@ function socketOnData(d) {
     }
     freeParser(parser, req, socket);
   } else if (parser.incoming && parser.incoming.complete &&
-             // When the status code is 100 (Continue), the server will
-             // send a final response after this client sends a request
-             // body. So, we must not free the parser.
-             parser.incoming.statusCode !== 100) {
+             // When the status code is informational (100, 102-199),
+             // the server will send a final response after this client
+             // sends a request body, so we must not free the parser.
+             // 101 (Switching Protocols) and all other status codes
+             // should be processed normally.
+             !informational(parser.incoming.statusCode)) {
     socket.removeListener('data', socketOnData);
     socket.removeListener('end', socketOnEnd);
     freeParser(parser, req, socket);
   }
 }
 
+// client
+function informational(status) {
```

Comment:
nit: name this `statusIsInformational` to be a bit more explicit.

Reply:
Done